react/components/Tabs/index.tsx

```diff
@@ -1,130 +1,39 @@
 
-interface HandleHideTabsInput {
-  tabs: HTMLCollection
-  tabsContainerWidth: number
-  tabsContainerFullWidth?: number
-  selectedTabIndex: number
-  tabsOrderList: number[]
+type Props = PropsWithChildren<{
   fullWidth?: boolean
-  isMobile?: boolean
-}
-
-interface HandleHideTabsOutput {
-  hideTabs: boolean
-  tabIndex: number
-}
-
-interface HandleShowSelectedHiddenTabInput extends HandleHideTabsOutput {
-  tabsOrderList: number[]
-  selectedTabIndex: number
-  setTabsOrderList: (input: number[]) => void
-}
+  sticky?: boolean
+}>
 
-const mapArrayToIndex: <T>(array: T[]) => number[] = array =>
-  array.map((_, index) => index)
-
-const handleHideTabs = ({
-  tabsContainerWidth,
-  tabsContainerFullWidth = 0,
-  tabs, // list with rendered tabs
-  selectedTabIndex,
-  tabsOrderList,
-  fullWidth,
-  isMobile,
-}: HandleHideTabsInput): HandleHideTabsOutput => {
-  let hideTabs = false
-  let tabIndex = 0
-  if (isMobile || fullWidth) {
-    // handle fullwidth
-    const numberOfTabs = tabsContainerFullWidth / DEFAULT_TAB_WIDTH
-    tabIndex = numberOfTabs - (numberOfTabs % 1)
-    hideTabs = tabIndex < tabsOrderList.length
-  } else {
-    const normalizedIndex = tabsOrderList.indexOf(selectedTabIndex)
-    let sumTabsWidth = tabs[normalizedIndex].clientWidth
-
-    // verify if is necessary hide tabs
-    for (; tabIndex < tabs.length; tabIndex++) {
-      const { clientWidth: childWidth } = tabs[tabIndex]
-      if (tabIndex !== normalizedIndex) {
-        sumTabsWidth += childWidth || DEFAULT_TAB_WIDTH
-      }
-
-      if (sumTabsWidth > tabsContainerWidth) {
-        hideTabs = true
-        if (tabIndex <= normalizedIndex) {
-          tabIndex++
-        }
-        break
-      }
-    }
-
-    // verify if the last tab can fit without more tabs button
-    if (
-      hideTabs &&
-      tabIndex + 1 === tabs.length &&
-      sumTabsWidth <= tabsContainerFullWidth
-    ) {
-      hideTabs = false
-      tabIndex = tabs.length
-    }
-  }
-
-  return { hideTabs, tabIndex }
-}
-
-const handleShowSelectedHiddenTab = ({
-  tabsOrderList,
-  hideTabs,
-  selectedTabIndex,
-  tabIndex,
-  setTabsOrderList,
-}: HandleShowSelectedHiddenTabInput): void => {
-  if (hideTabs && selectedTabIndex >= tabIndex) {
-    const leftList = tabsOrderList.slice(0, tabIndex - 1)
-    const rightList = tabsOrderList
-      .slice(tabIndex - 1)
-      .filter(i => i !== selectedTabIndex)
-    setTabsOrderList(leftList.concat([selectedTabIndex]).concat(rightList))
-  } else {
-    setTabsOrderList(tabsOrderList)
-  }
-}
-
-const propTypes = {
-  children: PropTypes.node,
-  fullWidth: PropTypes.bool,
-  isMobile: PropTypes.bool,
-  sticky: PropTypes.bool,
-}
-
-const Tabs: FC<InferProps<typeof propTypes>> = ({
-  children,
-  fullWidth,
-  isMobile,
-  sticky,
-}) => {
-  const childrenArray: Tab[] = Children.toArray(children)
+function Tabs({ children, fullWidth = false, sticky = false }: Props) {
+  const childrenArray = React.useMemo(
```

Comment:
children change every render, this may cause an overhead instead of help.

Reply:
not really. This is being used as a dependency on some effects. Without `useMemo` the effects will be called forever.